osu.Game.Rulesets.Mania/Mods/ManiaModScrollUp.cs

```diff
@@ -0,0 +1,100 @@
+    {
+        public override string Name => "Scroll up";
+        public override string Acronym => "SU";
+        public override LocalisableString Description => @"Key will become faster..., until you miss";
+        public override double ScoreMultiplier => 1;
+        public override IconUsage? Icon => FontAwesome.Solid.AngleDoubleUp;
+        public override ModType Type => ModType.Conversion;
+
+        private DrawableManiaRuleset drawableRuleset = null!;
+
+        private readonly BindableDouble scrollTime = new BindableDouble();
+
+        [SettingSource("Max Speed combo", "The combo count at which point the spead reaches max.")]
+        public BindableInt MaxComboCount { get; } = new BindableInt(480)
+        {
+            MinValue = 1,
+            MaxValue = 2000,
+        };
+
+        [SettingSource("Min Speed", "The speed when combo is 0", SettingControlType = typeof(SettingsSlider<double, ManiaScrollSlider>))]
+        public BindableDouble MinScoreSpeed { get; } = new BindableDouble(DrawableManiaRuleset.MAX_TIME_RANGE)
+        {
+            MinValue = DrawableManiaRuleset.MIN_TIME_RANGE,
+            MaxValue = DrawableManiaRuleset.MAX_TIME_RANGE,
+            Precision = 5,
+        };
+
+        [SettingSource("Max Speed", "The Max speed will reach", SettingControlType = typeof(SettingsSlider<double, ManiaScrollSlider>))]
+        public BindableDouble MaxScoreSpeed { get; } = new BindableDouble(DrawableManiaRuleset.MIN_TIME_RANGE)
+        {
+            MinValue = DrawableManiaRuleset.MIN_TIME_RANGE,
+            MaxValue = DrawableManiaRuleset.MAX_TIME_RANGE,
+            Precision = 5,
+        };
+
+        public ManiaModScrollUp()
+        {
+            MinScoreSpeed.BindValueChanged(s =>
+            {
+                MaxScoreSpeed.MaxValue = s.NewValue;
+            });
+
+            MaxScoreSpeed.BindValueChanged(s =>
+            {
+                MinScoreSpeed.MinValue = s.NewValue;
+            });
+        }
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<ManiaHitObject> drawableRuleset)
+        {
+            this.drawableRuleset = (DrawableManiaRuleset)drawableRuleset;
+        }
+
+        public void ApplyToPlayer(Player player)
+        {
+            drawableRuleset.ScoreSpeed.Disabled = false;
+            scrollTime.BindTo(drawableRuleset.ScoreSpeed);
+            scrollTime.Value = MinScoreSpeed.Value;
+        }
+
+        public void ApplyToScoreProcessor(ScoreProcessor scoreProcessor)
+        {
+            scoreProcessor.Combo.BindValueChanged(s =>
+            {
+                if (s.NewValue > MaxComboCount.Value)
+                {
+                    return;
+                }
+
+                double speed = MinScoreSpeed.Value - (MinScoreSpeed.Value - MaxScoreSpeed.Value) * Math.Log(s.NewValue + 1, MaxComboCount.Value + 1);
+
+                scoreProcessor.TransformBindableTo(scrollTime, speed, 500, Easing.OutQuint);
```

Comment:
inefficient, this is going to likely cause tens to hundreds of transforms to be running at any given time. i'm also not even sure that this is going to work correctly with multiple overlapping notes

Reply:
I refer to this https://github.com/ppy/osu/blob/b35e796d75153b53c1e450f407a48032aef1eddf/osu.Game/Rulesets/Mods/ModMuted.cs#L89
Because this value needs use transform instead of set directly, otherwise it will cause weird visual effects, and I didn't find a better solution either.
If there is a better way please let me know

Reply:
see mods that implement `IUpdatableByPlayfield`